server/src/test/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesResponseTests.java

```diff
@@ -57,4 +66,111 @@ public void testSerialization() throws IOException {
             assertEquals(deserialized.hashCode(), response.hashCode());
         }
     }
+
+    public void testToXContent() throws IOException {
+        FieldCapabilitiesResponse response = createSimpleResponse();
+
+        XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON)
+            .startObject();
+        response.toXContent(builder, ToXContent.EMPTY_PARAMS);
+        builder.endObject();
+
+        String generatedResponse = BytesReference.bytes(builder).utf8ToString();
+        assertEquals((
+            "{" +
+            "    \"fields\": {" +
+            "        \"rating\": { " +
+            "            \"keyword\": {" +
+            "                \"type\": \"keyword\"," +
+            "                \"searchable\": false," +
+            "                \"aggregatable\": true," +
+            "                \"indices\": [\"index3\", \"index4\"]," +
+            "                \"non_searchable_indices\": [\"index4\"] " +
+            "            }," +
+            "            \"long\": {" +
+            "                \"type\": \"long\"," +
+            "                \"searchable\": true," +
+            "                \"aggregatable\": false," +
+            "                \"indices\": [\"index1\", \"index2\"]," +
+            "                \"non_aggregatable_indices\": [\"index1\"] " +
+            "            }" +
+            "        }," +
+            "        \"title\": { " +
+            "            \"text\": {" +
+            "                \"type\": \"text\"," +
+            "                \"searchable\": true," +
+            "                \"aggregatable\": false" +
+            "            }" +
+            "        }" +
+            "    }" +
+            "}").replaceAll("\\s+", ""), generatedResponse);
+    }
+
+    private static FieldCapabilitiesResponse createSimpleResponse() {
+        Map<String, FieldCapabilities> titleCapabilities = new HashMap<>();
+        titleCapabilities.put("text", new FieldCapabilities("title", "text", true, false));
+
+        Map<String, FieldCapabilities> ratingCapabilities = new HashMap<>();
+        ratingCapabilities.put("long", new FieldCapabilities("rating", "long",
+            true, false,
+            new String[]{"index1", "index2"},
+            null,
+            new String[]{"index1"}));
+        ratingCapabilities.put("keyword", new FieldCapabilities("rating", "keyword",
+            false, true,
+            new String[]{"index3", "index4"},
+            new String[]{"index4"},
+            null));
+
+        Map<String, Map<String, FieldCapabilities>> responses = new HashMap<>();
+        responses.put("title", titleCapabilities);
+        responses.put("rating", ratingCapabilities);
+        return new FieldCapabilitiesResponse(responses);
+    }
+
+    public void testFromXContent() throws IOException {
```

Comment:
I think that we should make this test class extend `AbstractStreamableXContentTestCase`, then this test would come almost for free, as well as equals/hashcode tests and the existing serialization test.